crates/miden-tx/src/auth/tx_authenticator.rs

```diff
@@ -1,15 +1,55 @@
 use alloc::{collections::BTreeMap, string::ToString, sync::Arc, vec::Vec};
 
 use miden_objects::{
-    Felt, Word,
-    account::{AccountDelta, AuthSecretKey},
-    utils::sync::RwLock,
+    Felt, Hasher, Word, account::AuthSecretKey, crypto::SequentialCommit,
+    transaction::TransactionSummary, utils::sync::RwLock,
 };
 use rand::Rng;
 
 use super::signatures::get_falcon_signature;
 use crate::errors::AuthenticationError;
 
+// SIGNATURE DATA
+// ================================================================================================
+
+/// Data types on which a signature can be requested.
+///
+/// It supports three modes:
+/// - `TransactionSummary`: Structured transaction summary, recommended for authenticating
+///   transactions.
+/// - `Arbitrary`: Arbitrary payload provided by the application.
+/// - `Blind`: The underlying data is not meant to be displayed in a human-readable format. It can
+///   be assumed to already be a cryptographic commitment to some data.
+#[allow(clippy::large_enum_variant)]
+pub enum SignatureData {
```

Comment:
I wonder if there is a better way to name this than `SignatureData`. `SignatureData` makes it seems like this struct contains a signature, while what we want to say is that this is the data to be signed. The usual name in such cases is "message", but i'm not sure it would apply here (e.g., `Message` or `MessageDetails` sounds a bit too vague).

Reply:
I agree. Maybe `SigningInputs` is slightly better to signal that these are the inputs into the signing process. But I also don't like it too much, since it's still somewhat vague.

Reply:
I would stick with `SigningInputs` then, unless there are other suggestions